modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -244,6 +243,7 @@ public void processOpts() {
             // the "okhttp-gson" library template requires "ApiCallback.mustache" for async call
             supportingFiles.add(new SupportingFile("ApiCallback.mustache", invokerFolder, "ApiCallback.java"));
             supportingFiles.add(new SupportingFile("ApiResponse.mustache", invokerFolder, "ApiResponse.java"));
+            supportingFiles.add(new SupportingFile("JSON.mustache", invokerFolder, "JSON.java"));
```

Comment:
The PR modifes the Jersey2 client in a similar way, so the JSON is not needed in that case either. It is now only necessary for okhttp-gson, hence I moved it there in JavaCodegen.

I have regenerated both clients (jersey and jersey2) in the PR and all tests pass with the new clients. I had to remove the unit tests on JSON since that class no longer exists in jersey and jersey2 case.